ce_main_app/ikbd/ikbd_usb.cpp

```diff
@@ -399,52 +399,51 @@ void Ikbd::processMouse(input_event *ev)
     }
 }
 
-void Ikbd::processKeyboard(input_event *ev)
+void Ikbd::processKeyboard(input_event *ev, bool skipKeyboardTranslation)
 {
     int stKey = 0;
     int res;
 //    ikbdLog("processKeyboard");
 
     switch(ev->type) {
     case EV_SYN:
-        ikbdLog("Ikbd::processKeyboard() EV_SYN code=%04x\n", ev->code);
+        //ikbdLog("Ikbd::processKeyboard() EV_SYN code=%04x\n", ev->code);
         break;
     case EV_KEY:
-        ikbdLog("Ikbd::processKeyboard() EV_KEY code=%04x ev->value=%d\n", ev->code, ev->value);
+        //ikbdLog("Ikbd::processKeyboard() EV_KEY code=%04x ev->value=%d\n", ev->code, ev->value);
         statuses.ikbdUsb.aliveTime = Utils::getCurrentMs();
         statuses.ikbdUsb.aliveSign = ALIVE_KEYDOWN;
 
-        // ev->value -- 1: down, 2: auto repeat, 0: up
-        if(ev->code < KBD_KEY_COUNT) {
-            if(ev->value == 1) {    // key down
-                pressedKeys[ev->code] = true;
-                if(pressedKeys[KEY_LEFTCTRL] && pressedKeys[KEY_RIGHTCTRL] && pressedKeys[KEY_LEFTALT] && pressedKeys[KEY_RIGHTALT]) {
-                    toggleKeyboardExclusiveAccess();
-                }
-            } else if(ev->value == 0) { // key up
-                pressedKeys[ev->code] = false;
-            }
-        }
-
         stKey = keyTranslator.pcKeyToSt(ev->code);          // translate PC key to ST key
-
-        if(stKey == 0 || fdUart == -1) {                        // key not found, no UART open? quit
+        // ev->value -- 1: down, 2: auto repeat, 0: up
+        if(ev->value == 0) {        // when key is released, ST scan code has the highest bit set
+            stKey = stKey | 0x80;
+            pressedKeys[ev->code] = false;
+        } else if (ev->value == 2) {
+            // auto repeat has no place on Atari! :)
             return;
         }
 
-        if(keybJoy0 && keyJoyKeys.isKeybJoyKeyPc(0, ev->code)) {    // Keyb joy 0 is enabled, and it's a keyb joy 0 key? Handle it specially.
-            handlePcKeyAsKeybJoy(0, ev->code, ev->value);
-            return;
```

Comment:
Yep, my fault. I did see that but I was like whaaat, this is useless. OK, it is _not_ useless. ;)